# CONTENTS

- IMPORTING DATA
- EXPLORING DATA

# IMPORTING DATA

In [1]:
# import libraries
import pandas as pd
import numpy as np
import os 

In [2]:
# initialize path variable
path = r'/Users/smac_/Documents/courses/CF Data Analytics Program/Data Immersion/Achievement 4/05-2023 Instacart Basket Analysis'

In [3]:
# import data
vars_list = ['order_id', 'user_id', 'order_number', 'order_dow', 'order_hour_of_day', 'days_since_prior_order']
df_ords = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'orders.csv' ), usecols = vars_list)
df_prods = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'products.csv'), index_col = 0)

# EXPLORING DATA

## `'df_ords'` DATAFRAME

In [4]:
# first 5 rows
df_ords.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [5]:
# last 5 rows
df_ords.tail()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
3421078   2266710   206209            10          5                 18   
3421079   1854736   206209            11          4                 10   
3421080    626363   206209            12          1                 12   
3421081   2977660   206209            13          1                 12   
3421082    272231   206209            14          6                 14   

         days_since_prior_order  
3421078                    29.0  
3421079                    30.0  
3421080                    18.0  
3421081                     7.0  
3421082                    30.0

In [6]:
# dataframe size
df_ords.shape

(3421083, 6)

## `'df_prods'` DATAFRAME

In [7]:
# first 20 rows
df_prods.head(20)

product_name  aisle_id  \
product_id                                                                
1                                  Chocolate Sandwich Cookies        61   
2                                            All-Seasons Salt       104   
3                        Robust Golden Unsweetened Oolong Tea        94   
4           Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
5                                   Green Chile Anytime Sauce         5   
6                                                Dry Nose Oil        11   
7                              Pure Coconut Water With Orange        98   
8                           Cut Russet Potatoes Steam N' Mash       116   
9                           Light Strawberry Blueberry Yogurt       120   
10             Sparkling Orange Juice & Prickly Pear Beverage       115   
11                                          Peach Mango Juice        31   
12                                 Chocolate Fudge Layer Cake       119   
13                                          Saline Nasal Mist        11   
14                             Fresh Scent Dishwasher Cleaner        74   
15                                   Overnight Diapers Size 6        56   
16                              Mint Chocolate Flavored Syrup       103   
17                                          Rendered Duck Fat        35   
18                        Pizza for One Suprema  Frozen Pizza        79   
19           Gluten Free Quinoa Three Cheese & Mushroom Blend        63   
20             Pomegranate Cranberry & Aloe Vera Enrich Drink        98   

            department_id  prices  
product_id                         
1                      19     5.8  
2                      13     9.3  
3                       7     4.5  
4                       1    10.5  
5                      13     4.3  
6                      11     2.6  
7                       7     4.4  
8                       1     1.1  
9                      16     7.0  
10                      7     8.4  
11                      7     2.8  
12                      1     9.4  
13                     11     1.1  
14                     17     6.5  
15                     18    11.2  
16                     19     5.2  
17                     12    17.1  
18                      1    10.6  
19                      9    12.0  
20                      7     6.0

In [8]:
# last 35 rows
df_prods.tail(35)

product_name  aisle_id  \
product_id                                                                
49654                        Teriyaki Sauce, Sesame, Original         5   
49655                                             Apple Cider        98   
49656                              Masada Kosher Pocket Bread       128   
49657                                       Cabernet Tomatoes        83   
49658                              Brie with Herbs Foil Wedge         2   
49659                                 Organic Creamed Coconut        17   
49660                             Professionals Sleek Shampoo        22   
49661                                                   Porto       134   
49662                            Bacon Cheddar Pretzel Pieces       107   
49663       Ultra Protein Power Crunch Peanut Butter N' Ho...        57   
49664                          Lemon Cayenne Drinking Vinegar       100   
49665           Super Dark Coconut Ash & Banana Chocolate Bar        45   
49666                           Ginger Snaps Snacking Cookies        61   
49667                Enchilada with Spanish Rice & Beans Meal        38   
49668                          Apple Cinnamon Scented Candles       101   
49669                                        K Cup Dark Blend       100   
49670                                     Beef Summer Sausage       106   
49671                                    Milk Chocolate Drops        45   
49672                                  Cafe Mocha K-Cup Packs        26   
49673                   Stone Baked Multi Grain Artisan Rolls       129   
49674                 Frozen Greek Yogurt Bars Chocolate Chip        37   
49675                     Cinnamon Dolce Keurig Brewed K Cups        26   
49676                                  Ultra Red Energy Drink        64   
49677                         Thick & Chunky Sloppy Joe Sauce        59   
49678                         Large Chicken & Cheese Taquitos       129   
49679                                 Famous Chocolate Wafers        61   
49680                      All Natural Creamy Caesar Dressing        89   
49681                 Spaghetti with Meatballs and Sauce Meal        38   
49682                                      California Limeade        98   
49683                                          Cucumber Kirby        83   
49684               Vodka, Triple Distilled, Twist of Vanilla       124   
49685                      En Croute Roast Hazelnut Cranberry        42   
49686                                        Artisan Baguette       112   
49687              Smartblend Healthy Metabolism Dry Cat Food        41   
49688                                  Fresh Foaming Cleanser        73   

            department_id  prices  
product_id                         
49654                  13     4.0  
49655                   7    10.7  
49656                   3     7.1  
49657                   4     8.3  
49658                  16     3.9  
49659                  13     3.1  
49660                  11     6.7  
49661                   5     8.2  
49662                  19     3.6  
49663                  14    10.2  
49664                  21    13.7  
49665                  19     6.9  
49666                  19     5.2  
49667                   1     6.6  
49668                  17     5.6  
49669                  21     4.7  
49670                  12    19.2  
49671                  19     3.0  
49672                   7     6.5  
49673                   1     5.6  
49674                   1    11.1  
49675                   7    14.0  
49676                   7    14.5  
49677                  15     8.9  
49678                   1     3.4  
49679                  19     6.0  
49680                  13     4.9  
49681                   1     6.9  
49682                   7     4.3  
49683                   4    13.2  
49684                   5     5.3  
49685                   1     3.1  
49686                   3     7.8  
49687        

In [9]:
# column names
df_prods.columns

Index(['product_name', 'aisle_id', 'department_id', 'prices'], dtype='object')

In [10]:
# dataframe size
df_prods.shape

(49693, 4)

### WHAT IS THE MAXIMUM VALUE IN `'aisle_id'`?

In [11]:
# summary stats
df_prods.describe()

aisle_id  department_id        prices
count  49693.000000   49693.000000  49693.000000
mean      67.770249      11.728433      9.994136
std       38.316774       5.850282    453.519686
min        1.000000       1.000000      1.000000
25%       35.000000       7.000000      4.100000
50%       69.000000      13.000000      7.100000
75%      100.000000      17.000000     11.200000
max      134.000000      21.000000  99999.000000

The maximum value of `'aisle_id'` is 134.

### WHAT IS THE DATA TYPE OF `'department_id'`?

In [12]:
# dataframe info
df_prods.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 49693 entries, 1 to 49688
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   product_name   49677 non-null  object 
 1   aisle_id       49693 non-null  int64  
 2   department_id  49693 non-null  int64  
 3   prices         49693 non-null  float64
dtypes: float64(1), int64(2), object(1)
memory usage: 1.9+ MB


The data type of `'department_id'` column is int64 (INTEGER).